### ISSUE TITLE
PublishDir fail to move the output directory when the target is a S3 path

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#160]
For example: 

	process foo {
	  publishDir 's3://cbcrg-eu/work/hello/', mode: 'move'

	  output: 
	  file('x')

	  """
	  mkdir x 
	  touch x/one
	  touch x/two
	  """
	}

It create an empty `s3://cbcrg-eu/work/hello/x` path, instead it should contain the file `one` and `two`.

**Comment by jbyars** [comment]
Here is my example case. I'm probably just making a simple mistake. This is running on nextflow 0.18.3, ubuntu 14.04, aws region us-east-1, on r3.4xlarge instances. `mode: 'copy'` behaves as expected for me. `mode: 'move'` just creates empty folders in the bucket.
```
#!/usr/bin/env nextflow

/*
  This tests the behavior of publishDir outputting folders to buckets.
  Create a few txt files in the working folder before running this.
*/

params.bucket = "s3://nextflow-demo-output"

Channel
    .fromPath( "*.txt" )
    .set { copyfiles }

Channel
    .fromPath( "*.txt" )
    .set { movefiles }

process copytest {
    tag { sam }

    publishDir params.bucket + '/copy', mode: 'copy'

    input:
    file sam from copyfiles

    output:
    file out into copyresults

    script:
    out=sam.getBaseName()
    """
    mkdir "$out"
    touch "${out}/result1.txt"
    touch "${out}/result2.txt"
    touch "${out}/result3.txt"
    """
}

process movetest {
    tag { sam }

    publishDir params.bucket + '/move', mode: 'move'

    input:
    file sam from movefiles

    output:
    file out into copyresults

    script:
    out=sam.getBaseName()
    """
    mkdir "$out"
    touch "${out}/result1.txt"
    touch "${out}/result2.txt"
    touch "${out}/result3.txt"
    """
}
```

**Comment by pditommaso** [comment]
It's coherent with the test case that I've reported

**Comment by pditommaso** [comment]
Fixed in version `0.19.2`
